test/sqllogictest/subquery.slt

```diff
@@ -532,3 +532,65 @@ EXPLAIN PLAN FOR SELECT b != ALL(SELECT a FROM x) FROM y
 | Project (#3)
 
 EOF
+
+query T multiline
+EXPLAIN PLAN FOR SELECT b > ALL(SELECT a FROM x) FROM y
```

Comment:
Would you be up for decorating this with a comment about what it tests? For example, if in the future we improve this planning to be a `SELECT MAX(a) FROM x`, is that a bug in that this needs it to stay the way it is or what?

Reply:
Done. I think(?) we should be able to do that optimization just by peeking at this little bit here:
```
%3 =
| Join %1 %2
| | implementation = Differential %2 %1.()
| | demand = (#0, #1)
| Filter (#0 <= #1)
| Distinct group=(#0)
```

Reply:
It seems like it. However, I noticed after I typed that that it isn't a 100% win: the cross join is slow to start up, but then maintaining it is less stateful, I think. But I think you are right that just looking at the above we could make that transformation (though, it seems hard; at least, harder than reduction pushdown).

Reply:
Can you expand on what you mean by
> (though, it seems hard; at least, harder than reduction pushdown)

do you mean you think the better tack would be to plan with `min`/`max` from the get-go and then push that down?

Reply:
Oh I only meant that the "improved plan" would manufacture a reduction and push it down to the %2 input, which seemed to be harder that just the act of pushing a thing down. If we had
```
%3 =
| Join %1 %2
| | implementation = Differential %2 %1.()
| | demand = (#0, #1)
| Reduce group=(#0), func=(max(#1))
| Filter (#0 <= #1)
```
we would still have to learn to push down the reduction, I think.

Reply:
Perhaps missing context: we don't push down reductions yet. We should, but don't. Doing the transformation above probably goes in the queue behind that (in terms of complexity).